### ISSUE TITLE
Problem with package plugin

### ISSUE THREAD

**Issue by amateo** [ripienaar/mcollective-choria#83]
I have just install mcollective with choria.
I could find and ping nodes, but when I try to use the package plugin I get an error (other plugins like service does not work too).
In the client this is what I get:
```
amateo@joshua:~/puppetcode/mcollective$ mco rpc package status package=puppet-agent -I noctua10.um.es

 * [ ============================================================> ] 1 / 1


noctua10.um.es                           Unknown Request Status
   undefined method `properties' for nil:NilClass
```

and in the server log I have:
```
Dec 15 13:32:09 noctua10 mcollectived[12988]: choria.rb:147:in `validrequest?' Received valid request 6b342da51acc5efa901bb4fb61dde39a from choria=amateo.mcollective
Dec 15 13:32:09 noctua10 mcollectived[12988]: agent.rb:108:in `rescue in handlemsg' package#status failed: NoMethodError: undefined method `properties' for nil:NilClass
Dec 15 13:32:09 noctua10 mcollectived[12988]: agent.rb:109:in `rescue in handlemsg' /opt/puppetlabs/mcollective/plugins/mcollective/util/package/puppetpackage.rb:39:in `status'#012#011/opt/puppetlabs/mcollective/plugins/mcollective/agent/package.rb:105:in `do_pkg_action'#012#011/opt/puppetlabs/mcollective/plugins/mcollective/agent/package.rb:22:in `block in <class:Package>'#012#011/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/mcollective/rpc/agent.rb:86:in `handlemsg'#012#011/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/mcollective/agents.rb:126:in `block (2 levels) in dispatch'#012#011/opt/puppetlabs/puppet/lib/ruby/2.1.0/timeout.rb:90:in `block in timeout'#012#011/opt/puppetlabs/puppet/lib/ruby/2.1.0/timeout.rb:33:in `block in catch'#012#011/opt/puppetlabs/puppet/lib/ruby/2.1.0/timeout.rb:33:in `catch'#012#011/opt/puppetlabs/puppet/lib/ruby/2.1.0/timeout.rb:33:in `catch'#012#011/opt/puppetlabs/puppet/lib/ruby/2.1.0/timeout.rb:105:in `timeout'#012#011/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/mcollective/agents.rb:125:in `block in dispatch'
```

I have also tried running directly the agent, with the same result:
```
mateo@joshua:~/puppetcode/main$ mco package status puppet-agent -I noctua10.um.es

 * [ ==========================================================> ] 1 / 1

   noctua10.um.es: undefined method `properties' for nil:NilClass

Summary of Arch:

     No aggregate summary could be computed

Summary of Ensure:

     No aggregate summary could be computed


Finished processing 1 / 1 hosts in 21.42 ms
```

and the same log in the server.

These same commands works without problems in my previous platform configured with puppet/mcollective module instead of choria.

**Comment by ripienaar** [comment]
Strange. Does 'mco rpc rpcutil ping' work? This would then indicates a problem with the package agent specifically rather than a general problem. 

I will take a closer look but I apologise upfront you caught me mid house move without internet so maybe only next week

**Comment by amateo** [comment]
Yes, it works:
```
amateo@joshua:~/puppetcode/mcollective$ mco rpc rpcutil ping -I noctua10.um.es

 * [ ============================================================> ] 1 / 1

noctua10.um.es                           
   Timestamp: 1481806608

Finished processing 1 / 1 hosts in 20.61 ms
```

**Comment by ripienaar** [comment]
can you tell me exactly which distro and puppet version please? suspect its some incompatibility introduced in puppet but i couldnt reproduce here on what i have

**Comment by amateo** [comment]
Ubuntu 16.04 with puppet-agent 1.8.2-1.
But in the same node, with mcollective configured with puppet/mcollective the package plugin works.

**Comment by ripienaar** [comment]
It could be there is a mix of plugin versions if it was previously managed with that?

Mind pasting your server.cfg here and a find over all the directories mentioned in libdir in your confide?

**Comment by amateo** [comment]
This is my server.cfg:
```
main_collective = mcollective
collectives = mcollective

libdir = /opt/puppetlabs/mcollective/plugins

# consult the "classic" libdirs too
libdir = /opt/puppetlabs/mcollective/plugins
libdir = /opt/puppetlabs/mcollective/plugins

logfile = /var/log/puppetlabs/mcollective.log
loglevel = info
daemonize = 1

# Plugins
securityprovider = choria
plugin.psk = unset

connector = nats
plugin.activemq.pool.size = 1
plugin.activemq.pool.1.host = stomp1
plugin.activemq.pool.1.port = 6163
plugin.activemq.pool.1.user = mcollective
plugin.activemq.pool.1.password = marionette

# Facts
factsource = yaml
plugin.yaml = /etc/puppetlabs/mcollective/generated-facts.yaml
identity = noctua10.um.es
classesfile = /opt/puppetlabs/puppet/cache/state/classes.txt
rpcauthorization = 1
rpcauthprovider = action_policy
rpcaudit = 1
rpcauditprovider = logfile
plugin.rpcaudit.logfile = /var/log/puppetlabs/mcollective-audit.log
logger_type = syslog
```

**Comment by ripienaar** [comment]
Can confirm this is an interaction between choria and puppet due to a recently introduced feature:

```
irb(main):002:0> Puppet.settings.app_defaults_initialized?
=> nil
irb(main):003:0> Puppet.settings.preferred_run_mode = :agent
=> :agent
irb(main):004:0> Puppet.settings.initialize_global_settings([])
=> true
irb(main):005:0> Puppet.settings.initialize_app_defaults(Puppet::Settings.app_defaults_for_run_mode(Puppet.run_mode))
=> true
irb(main):006:0> Puppet::Type.type(:package).new({:name => "puppet-agent"}.merge({})).provider
Puppet::Error: Could not autoload puppet/type/package: Could not autoload puppet/provider/package/puppetserver_gem: no 'environments' in {:current_environment=><Puppet::Node::Environment:10368680 @name="*root*" @manifest="no_manifest" @modulepath="" >, :root_environment=><Puppet::Node::Environment:10368680 @name="*root*" @manifest="no_manifest" @modulepath="" >} at top of [[0, nil, nil]]
	from /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:55:in `lookup'
```

which eventually leads to what you see, I'll need to ping the Puppet devs to see what I am doing wrong

**Comment by ripienaar** [comment]
I released version `0.0.11` of the `mcollective_choria` module that resolves this

**Comment by amateo** [comment]
Thank you @ripienaar.
